pkg/process/util/address.go

```diff
@@ -2,12 +2,18 @@ package util
 
 import (
 	"encoding/binary"
+	"errors"
 	"net"
 )
 
+// ErrBufferTooSmall is returned when buffer is not large enough to hold
+// the byte representation of the IP address
+var ErrBufferTooSmall = errors.New("buffer not large enough to hold address")
+
 // Address is an IP abstraction that is family (v4/v6) agnostic
 type Address interface {
 	Bytes() []byte
+	Write([]byte) (int, error)
```

Comment:
I was a little confused by this. Usually `Write` signifies an input, i.e. the `[]byte` would be written _to_ the address. I think `WriteTo` would be more appropriate here. An `io.Writer` as input instead of `[]byte` would be ideal.

Reply:
You're right about that. That's was a confusing/poor choice of name. I'll change it